pydatastructs/strings/string_matching_algorithms.py

```diff
@@ -0,0 +1,136 @@
+from pydatastructs.linear_data_structures.arrays import (
+    OneDimensionalArray)
+
+__all__ = [
+    'find_string'
+]
+
+def find_string(text: str, pattern: str, algorithm: str) -> bool:
```

Comment:
The documentation for this should have the list of supported algorithms. For example,

https://github.com/codezonediitj/pydatastructs/blob/0dd2c03419f38e12983f6ec6e4220e4007c811a0/pydatastructs/graphs/algorithms.py#L270-L277

Full doc string of the above example is as follows,

https://github.com/codezonediitj/pydatastructs/blob/0dd2c03419f38e12983f6ec6e4220e4007c811a0/pydatastructs/graphs/algorithms.py#L260-L312

Adding note is optional in a doc string.